src/system/RoutesModule/Base/RoutesModuleInstaller.php

```diff
@@ -82,206 +79,205 @@ public function upgrade($oldVersion)
                 try {
                     $this->schemaTool->update($this->listEntityClasses());
                 } catch (\Exception $e) {
-                    if (System::isDevelopmentMode()) {
-                        $logger->error('{app}: Could not update the database tables during the upgrade. Error details: {errorMessage}.', ['app' => 'ZikulaRoutesModule', 'errorMessage' => $e->getMessage()]);
-
-                        return false;
-                    }
-                    $logger->error('{app}: Could not update the database tables during the ugprade. Error details: {errorMessage}.', ['app' => 'ZikulaRoutesModule', 'errorMessage' => $e->getMessage()]);
-
+                    $this->addFlash('error', $this->__('Doctrine Exception') . ': ' . $e->getMessage());
+                    $logger->error('{app}: Could not update the database tables during the upgrade. Error details: {errorMessage}.', ['app' => 'ZikulaRoutesModule', 'errorMessage' => $e->getMessage()]);
+    
                     return false;
                 }
         }
-
+    
         // Note there are several helpers available for making migration of your extension easier.
         // The following convenience methods are each responsible for a single aspect of upgrading to Zikula 1.4.0.
-
+    
         // here is a possible usage example
         // of course 1.2.3 should match the number you used for the last stable 1.3.x module version.
         /* if ($oldVersion = '1.2.3') {
             // update extension information about this app
             $this->updateExtensionInfoFor140();
-
+            
             // rename existing permission rules
             $this->renamePermissionsFor140();
-
+            
             // rename all tables
             $this->renameTablesFor140();
-
+            
             // remove event handler definitions from database
             $this->dropEventHandlersFromDatabase();
-
+            
             // update module name in the workflows table
             $this->updateWorkflowsFor140();
         } * /
     */
-
+    
         // update successful
         return true;
     }
-
+    
     /**
      * Renames this application in the core's extensions table.
      */
     protected function updateExtensionInfoFor140()
     {
         $conn = $this->getConnection();
         $dbName = $this->getDbName();
-
+    
```

Comment:
This is something I can not fix, caused by https://bugs.eclipse.org/bugs/show_bug.cgi?id=398172